src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCSVFileRecord.java

```diff
@@ -421,6 +421,40 @@ public Object[] getRowData() throws SQLServerException {
                             break;
                         }
 
+                        case microsoft.sql.Types.VECTOR: {
+                            int dimensionCount = cm.precision;
+                        
+                            if (dimensionCount <= 0) {
+                                throw new SQLServerException("Invalid vector dimension count.", null, 0, null);
+                            }
+                        
+                            String vectorData = data[pair.getKey() - 1].trim();
```

Comment:
pair.getKey() - 1 is repeated multiple times. Store it in a variable to improve readability and performance.